### PR TITLE
CORE-12311 Upsert users and fail on errors

### PR DESCRIPTION
**Pull request by davidcurrie**
Check whether the RBAC and crypto users exist before trying to create them. Add `-v ON_ERROR_STOP=1` to `psql` commands to fail on error.

### CODE REVIEW

File: charts/corda-lib/templates/_bootstrap.tpl [hunk 1]
```diff
@@ -51,18 +51,20 @@ spec:
...
+                DO \$\$ BEGIN IF EXISTS (SELECT FROM pg_catalog.pg_roles WHERE rolname = '$RBAC_DB_USER_USERNAME') THEN RAISE NOTICE 'Role "$RBAC_DB_USER_USERNAME" already exists'; ELSE CREATE USER "$RBAC_DB_USER_USERNAME" WITH ENCRYPTED PASSWORD '$RBAC_DB_USER_PASSWORD'; END IF; END \$\$;
+                GRANT USAGE ON SCHEMA {{ .Values.bootstrap.db.rbac.schema }} TO "$RBAC_DB_USER_USERNAME";
+                GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA {{ .Values.bootstrap.db.rbac.schema }} TO "$RBAC_DB_USER_USERNAME";
+                DO \$\$ BEGIN IF EXISTS (SELECT FROM pg_catalog.pg_roles WHERE rolname = '$CRYPTO_DB_USER_USERNAME') THEN RAISE NOTICE 'Role "$CRYPTO_DB_USER_USERNAME" already exists'; ELSE CREATE USER "$CRYPTO_DB_USER_USERNAME" WITH ENCRYPTED PASSWORD '$CRYPTO_DB_USER_PASSWORD'; END IF; END \$\$;
+                GRANT USAGE ON SCHEMA {{ .Values.bootstrap.db.crypto.schema }} TO "$CRYPTO_DB_USER_USERNAME";
+                GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA {{ .Values.bootstrap.db.crypto.schema }} TO "$CRYPTO_DB_USER_USERNAME";
+              SQL
```
Comment (davidcurrie) [inline charts/corda-lib/templates/_bootstrap.tpl:67]: Most of this change is just collapsing the multiple `psql` calls into one. The only changes within the SQL itself are the two `DO $$ BEGIN IF EXISTS...` around the `CREATE USER` calls, and the addition of the `-v ON_ERROR_STOP=1` option. The user of the HEREDOC requires the switch from `sh` to `bash`.

File: charts/corda-lib/templates/_bootstrap.tpl [hunk 2]
```diff
@@ -80,6 +82,7 @@ spec:
       {{- include "corda.bootstrapNodeSelector" . | nindent 6 }}
 
       restartPolicy: Never
+  backoffLimit: 0
```
Comment (davidcurrie) [inline charts/corda-lib/templates/_bootstrap.tpl:85]: Given we know that some of the init containers will fail if retried, set the limit to zero. This is also consistent with the other bootstrap jobs.

File: charts/corda-lib/templates/_bootstrap.tpl [hunk 3]
```diff
@@ -470,7 +473,7 @@ a second init container to execute the output SQL to the relevant database
...
-  command: [ 'sh', '-c',{{- if eq .name "db" }} 'for f in /tmp/*.sql; do psql -h {{ required "A db host is required" .Values.db.cluster.host }} -p {{ include "corda.clusterDbPort" . }} -f "$f" --dbname {{ include "corda.clusterDbName" . }}; done'{{- else }} 'psql -h {{ required "A db host is required" .Values.db.cluster.host }} -p {{ include "corda.clusterDbPort" . }} -f /tmp/{{ .sqlFile | default "db-config.sql" }} --dbname "dbname={{ include "corda.clusterDbName" . }} options=--search_path={{ .searchPath | default .Values.db.cluster.schema }}"' {{- end }} ]
+  command: [ 'sh', '-c', '-e',{{- if eq .name "db" }} 'for f in /tmp/*.sql; do psql -v ON_ERROR_STOP=1 -h {{ required "A db host is required" .Values.db.cluster.host }} -p {{ include "corda.clusterDbPort" . }} -f "$f" --dbname {{ include "corda.clusterDbName" . }}; done'{{- else }} 'psql -v ON_ERROR_STOP=1 -h {{ required "A db host is required" .Values.db.cluster.host }} -p {{ include "corda.clusterDbPort" . }} -f /tmp/{{ .sqlFile | default "db-config.sql" }} --dbname "dbname={{ include "corda.clusterDbName" . }} options=--search_path={{ .searchPath | default .Values.db.cluster.schema }}"' {{- end }} ]
```
Comment (davidcurrie) [inline charts/corda-lib/templates/_bootstrap.tpl:476]: Addition of `-e` on the shell and `-v ON_ERROR_STOP=1` on the `psql` commands to ensure we fail the job if the SQL fails.

**Review by oscardalmau-r3 — APPROVED** [review]
LGTM

**Pull request merged**